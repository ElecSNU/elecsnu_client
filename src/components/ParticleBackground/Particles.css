.particles {
    position: absolute;
    z-index: -1;
    width: 100%;
    height: 92.5%;
    overflow: hidden;
}
