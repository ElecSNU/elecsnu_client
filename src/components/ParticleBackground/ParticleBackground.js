import React from 'react';
import Particles from 'react-particles-js';

import './Particles.css';

const ParticleBackground = ({
    child: Component,
    ...rest
}) => {
    return (
        <>
            <Particles
                className='particles'
                params={{
                    background: {
                        color: {
                            // value: '#1e1b18',
                            value: '#fff',
                        },
                        // opacity: 0.3,
                    },
                    particles: {
                        color: {
                            value: '#d8315b',
                        },

                        number: {
                            density: {
                                enable: true,
                                value_area: 600,
                            },
                            value: 134,
                        },
                        size: {
                            value: 1,
                        },
                        collisions: {
                            enable: true,
                        },
                        links: {
                            color: '#0a2463',
                            distance: 80,
                            enable: true,
                            opacity: 0.8,
                            width: 1,
                        },
                    },
                    fpsLimit: 60,
                    detectRetina: true,
                }}
            />
            <Component />
        </>
    );
};

export default ParticleBackground;
