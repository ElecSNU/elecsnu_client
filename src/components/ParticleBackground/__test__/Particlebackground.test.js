  
   import { render, screen, fireEvent } from '@testing-library/react';
   import ReactDOM from 'react-dom'
   import React , { useState }from 'react'
   import renderer from 'react-test-renderer'
   import ParticleBackground from './../ParticleBackground';
   
   
   test('Tests the rendering of the page', () => {
       const div = document.createElement("div")
   const tree = renderer.create(<ParticleBackground child={"<index></index>"} ></ParticleBackground>).toJSON();
       expect(tree).toMatchSnapshot();
      });