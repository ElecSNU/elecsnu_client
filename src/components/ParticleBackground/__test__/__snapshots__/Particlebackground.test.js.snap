// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of the page 1`] = `
Array [
  <div
    className="particles"
    id="tsparticles"
  >
    <canvas
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>,
  <<index></index> />,
]
`;
